client/components/Home/Home.js

```diff
@@ -39,6 +50,21 @@ class Home extends Component {
   }
 
   render () {
+    const filteredProjects = this.props.projects.filter(project => {
+      let matched = true
+      this.state.filters.causes.forEach(cause => {
```

Comment:
what do you think of an approach like this:
```
this.props.projects.map(project => { if (
this.state.filters.causes.some(cause=> project.causes.indexOf(cause) >= 0) ){
return project; }
if(this.state.filters.technologies.some(tech => project.technologies.indexOf(tech) >=0){
return project; }
});
```

Reply:
This would be an OR set for all filters right?
I was going for an OR set for either the causes or technologies but AND set for both causes and technologies combined.
So a filter of "Animal" and "Javascript", the project would need both to show up.

Also, I get a blank page with undefined error when I substitute in your code. Is that because when the filter is empty, your code returns an empty array?
If that's the case we'll need to check the filters before applying .map().

I apologize if my solution is messy. I'm still quite inexperienced. Please help me to get better.

Reply:
this kind of logic should not be done in the render method. The render method should be the simplistic as possible, because it's a method that runs a lot of times (everytime a prop changes). My advice is that you do that on the handleCheckbox and have filteredprojects in state. If you want to do it in another way, you can use redux and save in store those filteredprojects and update them with an appropriate reducer.
if you want, talk to me in slack and we can go throught it :)